sumologic_collectd_metrics/metrics_converter.py

```diff
@@ -144,3 +155,82 @@ def convert_to_metrics(data, data_set, sep):
         metrics.append(metric)
 
     return metrics
+
+
+def process_signalfx_statsd_tags(data):
+    """
+    Process statsd tags embedded in the metric name within the supplied raw data. This function
+    modifies the raw data in-place.
+
+    :param data:
+    :type data: collectd.Values
+    :return:
+    :rtype: None
+    """
+    cleaned_type_instance, new_metadata = parse_statsd_signalfx_metric_name(
+        data.type_instance
+    )
+    data.type_instance = cleaned_type_instance
+    data.meta.update(new_metadata)
+
+
+def parse_statsd_signalfx_metric_name(metric_name):
+    """
+    Parse the metric name and extract metadata from it according to SignalFx' custom extension to
+    the StatsD protocol.
+    See: https://docs.signalfx.com/en/latest/integrations/agent/monitors/collectd-statsd.html#adding-dimensions-to-statsd-metrics
+
+    :param metric_name: The metric name
+    :type metric_name: str
+    :return: A pair consisting of the modified metric name without the metadata, and a dict of
+    extracted metadata.
+    :rtype: (str, dict)
+    """  # pylint: disable=line-too-long
+    matcher = re.compile(r"\[([^\]]+)\]")
+    matches = matcher.findall(metric_name)
+    if len(matches) > 1:
+        # if we find more than one metadata segment, something is probably wrong, and we should
+        # ignore it
+        logger.warning(
+            "Found two metadata segments in metric name `%s`. "
+            "This is likely a configuration error." % metric_name
+        )
+        return metric_name, {}
+
+    if len(matches) == 0:
+        # if we find no segments, we have nothing to do
+        return metric_name, {}
+
+    metadata = {}
+    tag_string = matches[0]
+    key_value_strings = tag_string.split(",")
+    for key_value_string in key_value_strings:
+        key_value_string = key_value_string.strip()
+        parts = [p.strip() for p in key_value_string.split("=")]
```

Comment:
At collectd level the format is inline with changes done here but the tags are in format `key:value` in sumo prod currently. 
Sample `gauge.edge-service.total_envs_per_cluster_by_stage[realm:prod,stage:dev,host_:1d6b250c-fe79-46ec-a254-54fa4b3c451e]" ` . 
Do check that once as well.

Reply:
Hm, is that why you wanted the separator to be configurable? Can you point me to any documentation of SignalFx that claims this is possible? I don't mind adding it, but I'd like some confirmation that it'll do what you need.

Reply:
Okay, so here is the thing which i found.
Collectd output: `statsd.[foo=bar,dim=val]test`
On Signalfx: `statsd.test  foo:bar  dim:val`
On sumo : `statsd.[foo:bar,dim:val]test`
I think internally all `=` gets converted to `:` both in Signalfx and Sumo, so the implementation  will work.

Reply:
Hm, we think this might be Sumo cleaning the metric name, this should work correctly if metrics use this format at the source.